### DISCUSSION TITLE
Add option to replace mouse events with pointer events in useHover hook

### DISCUSSION THREAD

**Discussion by rtivital** [floating-ui/floating-ui#1865] [React] [answered]
Disabled buttons do not support `mouseleave` event (https://github.com/facebook/react/issues/18753). This leads to several issues that occur with current implementation:

- Tooltip isn't shown when disabled button is hovered
- Tooltip will not be hidden if button is disabled while being hovered

Sandbox that illustrates both issues – https://codesandbox.io/s/divine-snow-48v1gl?file=/src/App.tsx

To solve these issues, I propose to add an option to `useHover` hook to use pointer events instead of mouse events:

```tsx
useHover({
  eventsType: 'pointer',
  // ... other options
});
```

**Comment by atomiks** [comment, marked as answer]
`pointerleave` doesn't fire on Safari (and those events have different hover semantics with touch input). Plus, it's still inaccessible via keyboard.

`disabled` prop should not be used at all if there is a tooltip attached to the button: https://floating-ui.com/docs/react-dom-interactions#disabled-elements

**Reply by zoubingwu** [reply]
> disabled prop should not be used at all

@atomiks  what if I want to have a tooltip to tell users why this button is disabled and how they should do. any recommandation?

**Reply by zoubingwu** [reply]
visually disabled still fires event when user click the button tho.

**Reply by zoubingwu** [reply]
I'm using a wrapper tag anyway, pm's needs must be satisfied 😊

**Reply by atomiks** [reply]
The link tells you what you should do. You early-return from event handlers (or conditionally set them), without using the `disabled` prop.
